packages/frontend/src/hooks/useVaultHistory.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useMemo, useState } from 'react'
+import { useQuery } from '@apollo/client'
+
+import { useWallet } from '@context/wallet'
+import { BIG_ZERO } from '@constants/index'
+
+import VAULT_HISTORY_QUERY, { VAULT_HISTORY_SUBSCRIPTION } from '../queries/squeeth/vaultHistoryQuery'
+import { VaultHistory, VaultHistoryVariables } from '../queries/squeeth/__generated__/VaultHistory'
+import { squeethClient } from '@utils/apollo-client'
+import { Action } from '@constants/index'
+import { toTokenAmount } from '@utils/calculations'
+import { useVaultManager } from '@hooks/contracts/useVaultManager'
+
+export const useVaultHistory = () => {
+  const { address, networkId } = useWallet()
+  const { vaults, firstValidVault } = useVaultManager()
+
+  const vaultId = vaults[firstValidVault]?.id ?? 0
+  const { data, subscribeToMore } = useQuery<VaultHistory, VaultHistoryVariables>(VAULT_HISTORY_QUERY, {
+    client: squeethClient[networkId],
+    fetchPolicy: 'cache-and-network',
+    variables: {
+      vaultId: vaultId,
+    },
+  })
+  const vaultHistory = data?.vaultHistories
+  useEffect(() => {
+    subscribeToMore({
+      document: VAULT_HISTORY_SUBSCRIPTION,
+      variables: {
+        vaultId: vaultId,
+      },
+      updateQuery(prev, { subscriptionData }) {
+        if (!subscriptionData.data) return prev
+        const newVaultsHistories = subscriptionData.data.vaultHistories
+        return { vaultHistories: newVaultsHistories }
+      },
+    })
+  }, [address, vaultId, subscribeToMore])
+
+  //accumulated four actions, mintedSqueeth doesn't take minted squeeth sold into account
```

Comment:
ok so mintedsqueeth is minted and not sold or LPed? and openShortSqueeth is any squeeth that was minted + short regardless if manual or through shortHelper?

Reply:
OpenShortSqueeth & mintedSqueeth are directly calculated from vault history, and so mintedSqueeth doesn't take sold or LPed into account

And mintedDebt is calculated from contract & swaps data
So I have finalOpenShort, compated mintedSqueeth & mintedDebt to get correct short amount including manually selling, if mintedSqueeth > mintedDebt means that users manually sell their minted oSQTH, so the openShort + mintedSqueeth - mintedDebt = actual short position